lib/shifts.rb

```diff
@@ -0,0 +1,66 @@
+require 'date'
+
+class Shifts
+
+    def initialize
+      @shifts = shifts
+      @keys = keys
+      @offsets = offsets
+    end
+
+    def random_number
+      rand(99999)
+    end
+
+    def padded_random_number
+      sprintf("%05d", random_number)
+    end
+
+    def random_number_array
+      padded_random_number.split(//).to_a
+    end
+
+    def key_pairs_array
+      pairs_array = []
+      random_number_array.each_cons(2) do |set|
+        pairs_array << set
+      end
+      pairs_array
+    end
+
+    def keys
+      {
+        "A key" => key_pairs_array[0].join.to_i,
+        "B key" => key_pairs_array[1].join.to_i,
+        "C key" => key_pairs_array[2].join.to_i,
+        "D key" => key_pairs_array[3].join.to_i
+        }
+    end
+
+    def todays_date
```

Comment:
Might have problems later with a preloaded date .... unsure of how that will work, so I plan to face that specific challenge when it arises.